lib/rules/init-declarations.js

```diff
@@ -49,25 +63,33 @@ module.exports = function(context) {
             for (var i = 0; i < declarations.length; ++i) {
                 var declaration = declarations[i],
                     id = declaration.id,
-                    initialized = isInitialized(declaration);
-
+                    initialized = isInitialized(declaration),
+                    isIgnoredForLoop = params.ignoreForLoops && isForLoop(node.parent);
                 if (id.type !== "Identifier") {
                     continue;
                 }
 
                 if (mode === MODE_ALWAYS && !initialized) {
                     context.report(declaration, "Variable '" + id.name + "' should be initialized on declaration.");
-                } else if (mode === MODE_NEVER && kind !== "const" && initialized) {
+                } else if (mode === MODE_NEVER && kind !== "const" && initialized && !isIgnoredForLoop) {
                     context.report(declaration, "Variable '" + id.name + "' should not be initialized on declaration.");
                 }
             }
         }
     };
-
 };
 
 module.exports.schema = [
     {
         "enum": ["always", "never"]
+    },
+    {
+        "type": "object",
+        "properties": {
```

Comment:
I think this should only be allowed for `never`, but shouldn't be allowed for `always`.